app/models/user.rb

```diff
@@ -92,9 +92,7 @@ def cohort
   end
 
   def school
-    return early_career_teacher_profile.school if early_career_teacher?
-    return mentor_profile.school if mentor?
-    return induction_coordinator_profile.schools.first if induction_coordinator?
+    teacher_profile&.school # POTENTIAL CHANGE OF BEHAVIOUR!
```

Comment:
Yeah this worries me a bit. I'd have to do some digging to be confident in this

Reply:
I've restored original implementation. One thing that is odd here is the scenario when SIC is also a mentor - would those two profile always belong to the same school?